### PR TITLE
adds updates to branch 6_getArticleComments

### PR DESCRIPTION
**Pull request by ericnabofa**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -103,4 +104,47 @@ describe('GET /api', () => {
...
+              comment_id: expect.any(Number),
+              votes: expect.any(Number),
+              created_at: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              article_id: 1,
+            });
+          }
+          const timestamps = comments.map(comment => new Date(comment.created_at).getTime());
+        expect(timestamps).toEqual([...timestamps].sort((a, b) => b - a));
```
Comment (itskatherine) [inline __tests__/app.test.js]: A good check
Might be easier to use a library like `jest-sorted`

File: __tests__/app.test.js [hunk 2]
```diff
@@ -103,4 +104,47 @@ describe('GET /api', () => {
...
+    test('GET:404 sends an appropriate status and error message when given a valid but non-existent id', () => {
+        return request(app)
+          .get('/api/articles/100/comments')
+          .expect(404)
+          .then((response) => {
+            expect(response.body.msg).toBe('comment does not exist');
+          });
+      });
+
+    test('GET:400 sends an appropriate status and error message when given an invalid id', () => {
```
Comment (itskatherine) [inline __tests__/app.test.js:142]: Clear test descriptions

File: __tests__/app.test.js [hunk 3]
```diff
@@ -103,4 +104,47 @@ describe('GET /api', () => {
...
+        return request(app)
+        .get('/api/articles/not-an-article/comments')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Bad Request')
+        })
+    });
+})
```
Comment (itskatherine) [inline __tests__/app.test.js:162]: Missing one more test, need a 200 test to check that when a request is made for an article with no comments, an empty array is returned.

**Review by itskatherine — CHANGES_REQUESTED** [review]
Hi Eric, good work so far,  could you please add the additional 200 tests and resubmit once you've got it passing

File: __tests__/app.test.js (hunk 1 above)
Comment (ericnabofa) [reply __tests__/app.test.js]: sure sure, i will update and use jest-sorted

File: __tests__/app.test.js (hunk 2 above)
Comment (ericnabofa) [reply __tests__/app.test.js:142]: thank you. you made it very clear from the get go in previous reviews 😊

File: __tests__/app.test.js (hunk 3 above)
Comment (ericnabofa) [reply __tests__/app.test.js:162]: alright. doing so now

also just about sending a pull request for task 7 😊

Comment (ericnabofa) [reply __tests__/app.test.js:162]: I did not realise this checking for correct respons will take my time. thanks katherine. Going back to solve this was really a eye opening.

Comment (ericnabofa) [reply __tests__/app.test.js:162]: submitting a fresh nchelp pr now

File: app/app.controller.js [hunk 4]
```diff
@@ -30,4 +31,21 @@ exports.getApiEndpoints = (req, res, next) => {
...
+}
+
+exports.getAllArticleComments = (req, res, next) => {
+    const {article_id} = req.params
+
+    const commentPromises = [selectCommentsByArticleId(article_id)]
+
+    if (article_id){
+        commentPromises.push(checkArticleExists(article_id))
+    }
```
Comment (itskatherine) [inline app/app.controller.js:43]: Great, looks good to go 👍

**Review by itskatherine — COMMENTED** [review]
Hi Eric, nice implementation of the new 200 test, good to merge 👍

File: app/app.controller.js (hunk 4 above)
Comment (ericnabofa) [reply app/app.controller.js:43]: Alrighty. Thanks
